### PR TITLE
Bugfix

### PR DESCRIPTION
**Pull request by yzhurko**

### CODE REVIEW

File: GUI/QtQuick/TxInputsModel.h [hunk 1]
```diff
@@ -125,7 +126,7 @@ class TxInputsModel : public QAbstractTableModel
    std::shared_ptr<spdlog::logger>  logger_;
    TxOutputsModel* outsModel_{ nullptr };
    const QMap<int, QString> header_;
-   std::map<bs::Address, std::vector<UTXO>>  utxos_;
+   std::map<bs::Address, std::set<UTXO>>  utxos_;
```
Comment (sergey-chernikov) [inline GUI/QtQuick/TxInputsModel.h]: UTXOs shouldn't be sorted, they should be in the order they come from the API.

File: GUI/QtQuick/QtQuickAdapter.cpp [hunk 2]
```diff
@@ -1439,6 +1439,8 @@ bool QtQuickAdapter::validateAddress(const QString& addr)
    const auto& addrStr = addr.toStdString();
    try {
       bs::Address::fromAddressString(addrStr);
+      return QRegularExpression(QString::fromLatin1("^(?:[13]{1}[a-km-zA-HJ-NP-Z1-9]{26,33}|bc1[a-z0-9]{39,59}|tb1[a-z0-9]{39,59})$")
+        ).match(addr).hasMatch();
```
Comment (sergey-chernikov) [inline GUI/QtQuick/QtQuickAdapter.cpp:1443]: Address is parsed in `fromAddressString`, based on the current mainnet/testnet selection, throwing an exception if parsing fails, you don't need to apply a regex on it.

File: GUI/QtQuick/TxInputsModel.cpp [hunk 3]
```diff
@@ -329,7 +329,7 @@ QUTXOList* TxInputsModel::getSelection()
    QList<QUTXO*> result;
    const double amount = outsModel_ ? outsModel_->totalAmount() : 0;
    logger_->debug("[{}] total amount: {}", __func__, amount);
-   if (amount > 0) { // auto selection
+   if (static_cast<int>(std::floor(amount * BTCNumericTypes::BalanceDivider)) > selectedBalance_) { // auto selection
```
Comment (sergey-chernikov) [inline GUI/QtQuick/TxInputsModel.cpp]: cast to int is wrong, BTC balance is always uint64_t or use BTCNumericTypes::satoshis

File: GUI/QtQuick/TxInputsModel.cpp [hunk 4]
```diff
@@ -374,6 +374,39 @@ QUTXOList* TxInputsModel::getSelection()
    return new QUTXOList(result, (QObject*)this);
 }
 
+void TxInputsModel::updateAutoselection()
+{
+    const double amount = outsModel_ ? outsModel_->totalAmount() : 0;
+
+    if (static_cast<int>(std::floor(amount * BTCNumericTypes::BalanceDivider)) <= selectedBalance_) {
```
Comment (sergey-chernikov) [inline GUI/QtQuick/TxInputsModel.cpp]: same casting type mismatch

File: GUI/QtQuick/TxInputsModel.h (hunk 1 above)
Comment (yzhurko) [reply GUI/QtQuick/TxInputsModel.h]: Tried to make vector of utxos unique using set; returned back to vector with insertion check

File: GUI/QtQuick/QtQuickAdapter.cpp (hunk 2 above)
Comment (yzhurko) [reply GUI/QtQuick/QtQuickAdapter.cpp:1443]: image
Then internal checks don't work - exceptions was not thrown for example for piece of code

**Review by sergey-chernikov — APPROVED** [review]

**Pull request merged**